### DISCUSSION TITLE
Speed up Text Extractor load in Obsidian

### DISCUSSION THREAD

**Discussion by hittjwX** [scambier/obsidian-text-extractor#65] [General]
`Text Extractor` with `Omnisearch` doubles the load time of my Obsidian startup. Any ideas to make it go faster? What about skipping extraction from files that have already been processed?

![image](https://github.com/user-attachments/assets/1dd19229-0ad3-443a-9470-04276cf9fc67)

**Comment by scambier** [comment]
> What about skipping extraction from files that have already been processed?

If that wasn't already the case, you'd still be waiting 😅 Files are really extracted once (on demand, not during startup), and then cached.

Though that's right it shouldn't take almost 3 seconds to load. On my dev vault, it "only" takes 400ms and even that is too long. I'll take a look at it.

**Reply by hittjwX** [reply]
I'm using git version control on my Obsidian vault. In most cases, Text-Extractor recreates cache files. I've not seen it delete them or modify previous ones. Maybe the opportunity is in how the cache works.

**Reply by hittjwX** [reply]
For example, when I move a set of folders containing images, Text-Extractor recreates cache files, but the filenames stay the same. 

Store the file name and path separately, then hash the filename. If the text doesn't change, don't recreate it; only modify the cache path.

Use an index of all the filenames to determine if a file is new, then create the cache for what's new. Use the index to check the cache, deleting what no longer represents a listed file.

It's the nature of text extraction to take up some processor. However, I'm working now from a beefy machine that still slows down noticeably when rebuilding the cache.
